lib/src/code/code.dart

```diff
@@ -46,17 +47,12 @@ class Code {
       singleLineCommentSequences: sequences,
     );
 
-    final serviceComments = _filterServiceComments(
+    final serviceComments = ServiceCommentFilter.filter(
       commentParser.comments,
       namedSectionParser: namedSectionParser,
     );
 
-    final lines = _textToCodeLines(
-      text: text,
-      highlighted: highlighted,
-      language: language,
-      commentsByLines: commentParser.getCommentsByLines(),
-    );
+    final serviceCommentsNodesSet = serviceComments.map((e) => e.source).toSet();
```

Comment:
If this is duplicated, is it better to make an extension method?

```dart
extension SingleLineCommentIterable on Iterable<SingleLineComment> {
  Set<Object?> get sources => {... map((e) => e.source)};
}
```

This can be in `single_line_comment.dart`.